### ISSUE TITLE
Bug: Found unparsable segment @L002P009: '(test.id) AS min_tes...'

### ISSUE THREAD

**Issue by mrshu** [alanmcruickshank/sqlfluff#171]
An interesting case with the `MIN` function (running on latest `master` branch):

```sql
SELECT
    MIN(test.id) AS min_test_id
FROM test
```

gets parsed without issue. When we add a space between `MIN` and `(test.id)`, however, we are met with the following error:

```
L:   2 | P:   9 |  PRS | Found unparsable segment @L002P009: '(test.id) AS min_tes...'
```

For completeness, here is what the query in question looks like:

```sql
SELECT
    MIN (test.id) AS min_test_id
FROM test
```

**Comment by alanmcruickshank** [comment]
Is it valid SQL without the space? At the moment the parser will fail to parse it because it thinks functions need to have the brackets straight after the name.

**Comment by mrshu** [comment]
@alanmcruickshank Running it on this db-fiddle.com (link here: https://www.db-fiddle.com/f/giavySG2eZqxuRa4ESgjyZ/1) suggests that it is -- at the very leas it runs in MySQL 5.7, PostgreSQL 12 and SQLite 3.30.

**Comment by alanmcruickshank** [comment]
Right you are - it's valid sql on my side. I'll implement this, but also with a linting rule to flag if the space is missing.
